cypress/integration/unit_tests/update_test.js

```diff
@@ -1,137 +1,118 @@
   loadScriptsBeforeForUnitTests('firebase', 'jquery');
 
-  before(() => {
-    global.google = {maps: {event: {clearListeners: () => {}}}};
-
-    const formDiv = document.createElement('div');
-    formDiv.class = 'form';
-    formDiv.id = 'form-div';
-
-    document.body.appendChild(formDiv);
-
-    const snackbarDiv = document.createElement('div');
-    snackbarDiv.id = 'snackbar';
-    const snackbarText = document.createElement('span');
-    snackbarText.id = 'snackbar-text';
-    snackbarDiv.appendChild(snackbarText);
-    document.body.appendChild(snackbarDiv);
-  });
+  before(() => global.google = {maps: {event: {clearListeners: () => {}}}});
 
   // creates the form div and stubs the relevant document methods.
   beforeEach(() => {
-    // function that resolves the createAndDisplayJoinedDataPromise
-    let resolvePromise;
-    createAndDisplayJoinedDataPromise =
-        new Promise((resolve) => resolvePromise = resolve);
-    createAndDisplayJoinedDataStub =
-        cy.stub(Run, 'createAndDisplayJoinedData')
-            .callsFake((_, valuesPromise) => {
-              valuesPromise.then((toggles) => {
-                lastPassedPovertyThreshold = toggles.povertyThreshold;
-                lastPassedDamageThreshold = toggles.damageThreshold;
-                lastPassedPovertyWeight = toggles.povertyWeight;
-                resolvePromise();
-              });
-            });
-    cy.stub(LayerUtil, 'removeScoreLayer');
-
-    lastPassedDamageThreshold = null;
-    lastPassedPovertyThreshold = null;
-    lastPassedPovertyWeight = null;
-
-    $('#form-div').empty();
+    cy.wrap(cy.stub(Run, 'createAndDisplayJoinedData')).as(
+        'createAndDisplayJoinedDataStub');
+    cy.wrap(cy.stub(LayerUtil, 'removeScoreLayer')).as('removeScoreLayerStub');
+
+    cy.visit('test_utils/empty.html');
+    cy.document().then((doc) => {
+      const formDiv = doc.createElement('div');
+      formDiv.id = 'form-div';
+      doc.body.appendChild(formDiv);
+      cy.stub(document, 'getElementById')
+          .callsFake((id) => doc.getElementById(id));
+    });
   });
 
-  afterEach(() => $('#form-div').empty());
-
   it('does not have a damage asset', () => {
     const nullData = {asset_data: {damage_asset_path: null}};
-    setUpToggles(Promise.resolve({data: () => nullData}), {}).then(() => {
-      expect($('input').length).to.equal(3);
-
-      $('[id="poverty threshold"]').val(0.05);
-      $('#update').trigger('click');
-
-      expect(createAndDisplayJoinedDataStub).to.be.calledOnce;
-
-      expect(toggles.get('poverty weight')).to.equals(1);
-      expect($('#error').text()).to.equal('');
-      cy.wrap(createAndDisplayJoinedDataPromise).then(() => {
-        expect(lastPassedPovertyWeight).to.equals(1);
-        expect(lastPassedDamageThreshold).to.equals(0.0);
-      });
-    });
+    cy.wrap(setUpToggles(Promise.resolve({data: () => nullData}), {}));
+    cy.get('input').should('have.length', 2);
```

Comment:
Ah is this from current settings button getting deleted

Reply:
Oh, that makes sense.